sdk/src/account.rs

```diff
@@ -79,6 +79,20 @@ impl From<Account> for AccountSharedData {
 
 pub trait WritableAccount: ReadableAccount {
     fn set_lamports(&mut self, lamports: u64);
+    fn add_lamports(&mut self, lamports: u64) {
```

Comment:
How much pain would it be to make these fallible?

We have friends that like it when we pay them to find paths to panics :wink:

Reply:
I discussed with @jackcmay in slack. I will try returning Result<> or Option<> here and the caller can decide what to do.

Reply:
program calles can return illegal instruction errors